File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.swift
import Foundation

func solution(_ ingredient:[Int]) -> Int {
    // 1,2,3,1 순서로 나와야하네.
    var result = 0
    var ary1 : [Int] = []
    
    for i in ingredient {
        ary1.append(i)
        let last = ary1.suffix(4)
        if last == [1,2,3,1] {
            result += 1
            ary1.removeLast(4)
        }
    }
    return result
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    const ary = []
    let result = 0
    
    for(let i = 0; i<ingredient.length; i++) {
        ary.push(ingredient[i])
        
        const value = ary.slice(-4).join("")
        
        if(value === "1231"){
            result += 1
            ary.splice(-4,4)
        }
    }
    
    return result
    
    
}